### PR TITLE
Form and Trigger Automation Tests

### PR DESCRIPTION
**Pull request by labkey-sweta**
#### Rationale
Automation test for https://docs.google.com/document/d/14OgIGqOSTUpQK9lyeebHrc_4i36cYDci-V7QLSlvM6o/edit

#### Related Pull Requests
* 

#### Changes
*

### CODE REVIEW

File: onprc_ehr/test/src/org/labkey/test/tests/onprc_ehr/ONPRC_EHRTest2.java [hunk 1]
```diff
@@ -967,6 +968,67 @@ public void testManageCases() throws Exception
...
     }
 
+    @Test
+    public void testFormTriggers()
+    {
+        String aliveAnimal = "99998";
+        String deadAnimal = "99999"; //Mark an animal dead.
+
+        Map<String, Map<String, String>> formsToTest = Map.of(
+                "Birth", Map.of("allowDatesInDistantPast", "true", "allowFutureDates", "true", "allowDeadIds", "true", "allowAnyId", "true"),
```
Comment (labkey-martyp) [inline onprc_ehr/test/src/org/labkey/test/tests/onprc_ehr/ONPRC_EHRTest2.java]: Can you add another property to the inner maps, gridToTest which will be the grid in the form to test.

File: onprc_ehr/test/src/org/labkey/test/tests/onprc_ehr/ONPRC_EHRTest2.java [hunk 2]
```diff
@@ -967,6 +968,67 @@ public void testManageCases() throws Exception
...
+        clickAndWait(Locator.linkWithText("Enter Data / Task Review"));
+        for (Map.Entry<String, Map<String, String>> form : formsToTest.entrySet())
+        {
+            clickAndWait(Locator.linkWithText(form.getKey()));
+            enableForm();
+            Ext4GridRef grid;
+            if(form.getKey().equals("Housing Transfers"))
+                grid = _helper.getExt4GridForFormSection(form.getKey());
+            else
+                grid = _helper.getExt4GridForFormSection(form.getKey() + "s");
```
Comment (labkey-martyp) [inline onprc_ehr/test/src/org/labkey/test/tests/onprc_ehr/ONPRC_EHRTest2.java]: Like I mentioned above, lets just specify in the initial table what grid to test.

File: onprc_ehr/test/src/org/labkey/test/tests/onprc_ehr/ONPRC_EHRTest2.java [hunk 3]
```diff
@@ -967,6 +968,67 @@ public void testManageCases() throws Exception
...
+            _helper.addRecordToGrid(grid);
+            if (form.getValue().get("allowDeadIds").equals("true"))
```
Comment (labkey-martyp) [inline onprc_ehr/test/src/org/labkey/test/tests/onprc_ehr/ONPRC_EHRTest2.java:1007]: Can we test the else on these as well. If the value is false.

File: onprc_ehr/test/src/org/labkey/test/tests/onprc_ehr/ONPRC_EHRTest2.java (hunk 1 above)
Comment (labkey-sweta) [reply onprc_ehr/test/src/org/labkey/test/tests/onprc_ehr/ONPRC_EHRTest2.java]: Done

File: onprc_ehr/test/src/org/labkey/test/tests/onprc_ehr/ONPRC_EHRTest2.java (hunk 2 above)
Comment (labkey-sweta) [reply onprc_ehr/test/src/org/labkey/test/tests/onprc_ehr/ONPRC_EHRTest2.java]: Done

**Review by labkey-martyp — APPROVED** [review]

**Pull request merged**